neo/Consensus/ConsensusContext.cs

```diff
@@ -137,21 +137,23 @@ public bool Load()
 
         public ConsensusPayload MakeChangeView(byte newViewNumber)
         {
+            var changeViewLockedCount = ChangeViewPayloads.Count(p => p != null && p.GetDeserializedMessage<ChangeView>().NewViewNumber > ViewNumber);
+            if (ChangeViewPayloads[MyIndex]?.GetDeserializedMessage<ChangeView>().NewViewNumber <= ViewNumber && newViewNumber > ViewNumber)
+                changeViewLockedCount++;
             return ChangeViewPayloads[MyIndex] = MakeSignedPayload(new ChangeView
             {
                 NewViewNumber = newViewNumber,
-                Timestamp = TimeProvider.Current.UtcNow.ToTimestamp()
+                Timestamp = TimeProvider.Current.UtcNow.ToTimestamp(),
+                Locked = changeViewLockedCount >= this.M() && !this.ResponseSent() && !(this.IsPrimary() && this.RequestSentOrReceived())
```

Comment:
In this locked correct? is'nt it || insteand of &&?

Reply:
No it is correct as is. I think i can simplify the last part to just !this.IsPrimary(), because primary should always have sent or received the prepare request by the time it would time out and send a change view.

Reply:
Oh you are right. this logic needs an adjustment!

Reply:
Should be fixed. Have another look.